event_sale_reservation/wizards/registration_editor.py

```diff
@@ -37,6 +37,7 @@ def action_convert_to_registration(self):
         "incompatible with each other"."""
         # Modify SO lines to be tickets instead of reservations
         for line in self.event_registration_ids:
+            line.sale_order_line_id.product_updatable = True
```

Comment:
Why not put it in the following `write`?

Reply:
Because validation is done before applying the write:
https://github.com/odoo/odoo/blob/17.0/addons/sale/models/sale_order_line.py#L1009-L1011

Reply:
The error does not appear in the tests because the same product was being used for reservations and registrations. Secondly, creating the order with create is not the most accurate functional test, and thirdly, in the test environment, the order of execution of the computations is not the same as in a production environment. Even so, I have incorporated the most accurate tests so that they can also serve as documentation of the problem.